media-proxy/src/mesh/st2110.cc

```diff
@@ -0,0 +1,153 @@
+#include "st2110.h"
+
+namespace mesh::connection {
+
+st_frame_fmt ST2110::mesh_video_format_to_st_format(int fmt)
+{
+    switch (fmt) {
+    case MESH_VIDEO_PIXEL_FORMAT_NV12:
+        return ST_FRAME_FMT_YUV420CUSTOM8;
+    case MESH_VIDEO_PIXEL_FORMAT_YUV422P:
+        return ST_FRAME_FMT_YUV422PLANAR8;
+    case MESH_VIDEO_PIXEL_FORMAT_YUV422P10LE:
+        return ST_FRAME_FMT_YUV422PLANAR10LE;
+    case MESH_VIDEO_PIXEL_FORMAT_YUV444P10LE:
+        return ST_FRAME_FMT_YUV444PLANAR10LE;
+    case MESH_VIDEO_PIXEL_FORMAT_RGB8:
+        return ST_FRAME_FMT_RGB8;
+    default:
+        return ST_FRAME_FMT_YUV422PLANAR10LE;
+    }
+}
+
+st30_fmt ST2110::mesh_audio_format_to_st_format(int fmt)
```

Comment:
Rework this to return an error if the format is unknown.

Reply:
Fixed in cbda05db27e270e513ce1c8d50c47fd4da7ba651